pkg/api/request/request.go

```diff
@@ -37,8 +38,8 @@ type Request struct {
 	err           error
 }
 
-func NewRequest(requestType string, uri string) *Request {
-	return &Request{requestType: requestType, uri: uri}
+func NewRequest(requestType string, uri string, userAgent string) *Request {
```

Comment:
This seems like two concepts in place here, part of attributes in constructor and part in separate with methods method - builder pattern. Why dont we unify this approaches and use only builders, similar to JAX-RS usage?